src/Components/Components/src/Microsoft.AspNetCore.Components.csproj

```diff
@@ -17,19 +17,6 @@
     <Reference Include="Microsoft.Extensions.DependencyInjection.Abstractions" />
   </ItemGroup>
 
-  <!--
-    These references exist only in the .nuspec files and baseline checks are not aware of them. Keep suppressions
-    in sync with the two .nuspec files:
-    - Anytime a reference is added to this project file, remove its suppression.
-    - Remove InNuspecFile attributes of references removed from the .nuspec files. Make suppression conditional on
-      Major.Minor during previews. After RTM (and baseline updates), remove suppressions entirely.
-  -->
-  <ItemGroup>
-    <SuppressBaselineReference Include="Microsoft.AspNetCore.Components.Analyzers" InNuspecFile="true" />
-    <SuppressBaselineReference Include="Microsoft.AspNetCore.Authorization" InNuspecFile="true" />
-    <SuppressBaselineReference Include="Microsoft.JSInterop" Condition=" '$(AspNetCoreMajorMinorVersion)' == '6.0' " />
-  </ItemGroup>
```

Comment:
Very surprised you can remove all of this because the `InNuspecFile` dependencies still exist only in the .nuspec file.

The problem appears to be a fix from a servicing branch hasn't been brought forward into `main`. This block
``` xml
  <ItemGroup Condition=" '$(PackageId)' == 'Microsoft.AspNetCore.Components' AND '$(TargetFramework)' == 'net50' ">
    <BaselinePackageReference Include="Microsoft.AspNetCore.Components.Analyzers" Version="[5.0.2, )" />
    <BaselinePackageReference Include="Microsoft.AspNetCore.Authorization" Version="[5.0.2, )" />
  </ItemGroup>
```
no-ops incorrectly.

The bug is in the BaselineGenerator. Calculation of `defaultTarget` is based on the `Version` attribute (`5.0`) in the file (see https://github.com/dotnet/aspnetcore/blob/1a2723ded40a4204cb7b7f262b0e1388d7607fde/eng/tools/BaselineGenerator/Program.cs#L100-L105) and that means nothing handles `net6.0`.

Reply:
`OR` clauses should be inserted by https://github.com/dotnet/aspnetcore/blob/1a2723ded40a4204cb7b7f262b0e1388d7607fde/eng/tools/BaselineGenerator/Program.cs#L178-L185

Reply:
The `BaselineGenerator` didn't get all of the `Condition`s right in eng/Baseline.Designer.props e.g.

``` xml
  <ItemGroup Condition=" '$(PackageId)' == 'Microsoft.AspNetCore.Components' AND '$(TargetFramework)' == 'net50' ">
    <BaselinePackageReference Include="Microsoft.AspNetCore.Components.Analyzers" Version="[5.0.4, )" />
    <BaselinePackageReference Include="Microsoft.AspNetCore.Authorization" Version="[5.0.4, )" />
  </ItemGroup>
```

It's not clear to me where the invalid TFM `net50` is coming from. Might be something in the NuGet packages we're using in the `BaselineGenerator`. That string doesn't appear in our 'main' or 'release/5.0' branches. Could you debug a bit❔

There are 26 instances of this `Condition` in eng/Baseline.Designer.props and it's not clear what ties the affected packages together either.

Reply:
@dougbu assuming you mean `net50` as opposed to `net5.0`? Checking it out now

Reply:
ok, when we're seeing `net50`, `group.TargetFramework` = `{.NETFramework,Version=v5.0}`. When we're seeing `net5.0`, `group.TargetFramework` = `{.NETCoreApp,Version=v5.0}`. Looks like the difference is coming from the .nuspec's. When we get the desired behavior (`group.TargetFramework` = `{.NETCoreApp,Version=v5.0}`), the .nuspec has the following:

`<group targetFramework=".NETCoreApp5.0">`

When we get `group.TargetFramework` = `{.NETFramework,Version=v5.0}`, the .nuspec has:

`<group targetFramework="net5.0">`

So it looks like the nuget code we're using interprets `net5.0` as .Net Framework 5, which it converts to `net50` via `GetShortFolderName()` - whereas it treats `netcoreapp5.0` as one would expect. I don't know why our nuspec's have different versions of the net50 TFM per package, but short of fixing that in servicing, we could just account for this in the baseline generator by treating both `net5.0` and `net50` as matches in https://github.com/dotnet/aspnetcore/blob/1a2723ded40a4204cb7b7f262b0e1388d7607fde/eng/tools/BaselineGenerator/Program.cs#L178-L185

Side note, does the above behavior surprise you, or is it expected @nkolev92 @dsplaisted

Reply:
As far as I can tell, the difference on our end is the use of `TargetFramework` vs. `TargetFrameworks`. Projects that use `TargetFramework` get `NetCoreApp5.0` in the .nuspec, projects that use `TargetFrameworks` get `net5.0`:

https://github.com/dotnet/aspnetcore/blob/19785ccdf595772231ee7b93083471ff13bad3de/src/Azure/AzureAppServicesIntegration/src/Microsoft.AspNetCore.AzureAppServicesIntegration.csproj#L5

https://github.com/dotnet/aspnetcore/blob/19785ccdf595772231ee7b93083471ff13bad3de/src/Identity/ApiAuthorization.IdentityServer/src/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.csproj#L4

https://github.com/dotnet/aspnetcore/blob/19785ccdf595772231ee7b93083471ff13bad3de/src/Azure/AzureAppServices.HostingStartup/src/Microsoft.AspNetCore.AzureAppServices.HostingStartup.csproj#L7

https://github.com/dotnet/aspnetcore/blob/19785ccdf595772231ee7b93083471ff13bad3de/src/Security/Authorization/Core/src/Microsoft.AspNetCore.Authorization.csproj#L8

Again, is that expected @nkolev92 @dsplaisted? This one is particularly surprising

Reply:
This behavior isn't present in 6.0 - all of these packages have `net6.0` in the .nuspec there

Reply:
The present behaviour feels like significant problems in both the `pack` code (writing out an unexpected TFM) and NuGet.Packaging (misinterpreting the TFM as .NET Framework). @rrelyea this a NuGet bug that's been fixed in another 5.0 .NET SDK or NuGet package❔

We currently use the 5.0.104 SDK and NuGet.Versioning v5.7.0 in our release/5.0 branch, where these packages were written and where the baseline file was generated.

Reply:
I think the problem is that NuGet.Versioning 5.7.0 may not know about or handle the .NET 5 target frameworks correctly.  I believe NuGet.Versioning 5.8.0 is the one that corresponds with .NET 5 / .NET SDK 5.0.100.

Also adding @zkat.

Reply:
Oops, I forgot our `BaselineGenerator` tool uses an explicit `@(PackageReference)` item. The tool references NuGet.Protocol v4.8.2. Probably need v5.8.1 both here and in the 'release/5.0' branch.

Reply:
We should also backport https://github.com/dotnet/aspnetcore/pull/31064 to 5.0 so the .nuspec's get fixed

Reply:
@wtgodbe I'm not sure that'll help because it doesn't touch the `BaselineGenerator` tool or the SDK (don't think `pack` uses the package versions specified in this repo). I updated the tool here and am testing to see whether it now generates the expected `Condition`s despite the .nuspec inconsistency.

Reply:
But, it won't hurt❕

Reply:
You're right, https://github.com/dotnet/aspnetcore/pull/31064 wasn't what fixed the .nuspec's in `main`. They were fixed in preview2 (the preview1 .nuspec's still have the weird TFMs), and my change was merged in the preview3 time frame.

Reply:
Maybe this? https://github.com/dotnet/aspnetcore/commit/e812aee11957c69c1cb95121fa793ed0a09b543b

Updated `NuGetVersioningVersion` from 5.7.0 to 5.9.0-preview

Reply:
> We should also backport #31064 to 5.0 so the .nuspec's get fixed

Unfortunately you were correct @wtgodbe. We need to correct the 5.0 packages before the updated tool can generate Baseline.Designer.props with the right `Condition`s. Could edit that file manually if you want to get this in first.

Probably a good idea to take the `BaselineGenerator.csproj` change I made back to release/5.0 too.

@dsplaisted @zkat exactly what should we update to ensure the 5.0 packages are consistently created with the expected `<group targetFramework=".NETCoreApp5.0">` in the .nuspec files❔ (@wtgodbe's point about `@(TargetFramework)` versus `@(TargetFrameworks)` is also concerning.)

Reply:
To be more clear, running the tool in the updated wtgodbe/BaselineSuppress branch (which contains both Pranav's commit and your #31064 update) doesn't fix what gets generated.

Reply:
For now we could update the condition like I suggested in https://github.com/dotnet/aspnetcore/pull/29934#discussion_r606337886, which I believe should cause the generator to create the file how we want it to. If we update the nuget versions in 5.0 now we'll still have to wait until May before we can get packages with correct .nuspec's

Reply:
https://github.com/dotnet/aspnetcore/pull/31490 should cover the 5.0 part (I'll check the .nuspec's once the build is done)

Reply:
> @dsplaisted @zkat exactly what should we update to ensure the 5.0 packages are consistently created with the expected `<group targetFramework=".NETCoreApp5.0">` in the .nuspec files❔ (@wtgodbe's point about `@(TargetFramework)` versus `@(TargetFrameworks)` is also concerning.)

I think it should all work correctly if you are using the .NET 5.0 tooling, and referencing the right versions of the NuGet libraries.

Reply:
> we could update the condition like I suggested in https://github.com/dotnet/aspnetcore/pull/29934#discussion_r606337886

If you mean
> account for this in the baseline generator

I echoed that in https://github.com/dotnet/aspnetcore/pull/29934#discussion_r606396738 and your timing point makes sense
> Could edit that file manually if you want to get this in first.

Reply:
The changes to `BaselineGenerator` look great❕ If you need to revert the Microsoft.AspNetCore.Components.Analyzers" InNuspecFile and Microsoft.AspNetCore.Authorization suppressions, we're good to go.